dask_cloudprovider/providers/azure/azureml.py

```diff
@@ -0,0 +1,751 @@
+            cores_or_gpus,
+            cores,
+            memory,
+        )
+        return text
+
+    def _widget(self):
+        """ Create IPython widget for display within a notebook """
+        try:
+            return self._cached_widget
+        except AttributeError:
+            pass
+
+        try:
+            from ipywidgets import Layout, VBox, HBox, IntText, Button, HTML, Accordion
+        except ImportError:
+            self._cached_widget = None
+            return None
+
+        layout = Layout(width="150px")
+
+        if self.dashboard_link:
+            dashboard_link = (
+                '<p><b>Dashboard: </b><a href="%s" target="_blank">%s</a></p>\n'
+                % (self.dashboard_link, self.dashboard_link,)
+            )
+        else:
+            dashboard_link = ""
+
+        if self.jupyter_link:
+            jupyter_link = (
+                '<p><b>Jupyter: </b><a href="%s" target="_blank">%s</a></p>\n'
+                % (self.jupyter_link, self.jupyter_link,)
+            )
+        else:
+            jupyter_link = ""
+
+        title = "<h2>%s</h2>" % self._cluster_class_name
+        title = HTML(title)
+        dashboard = HTML(dashboard_link)
+        jupyter = HTML(jupyter_link)
+
+        status = HTML(self._widget_status(), layout=Layout(min_width="150px"))
+
+        if self._supports_scaling:
+            request = IntText(
+                self.initial_node_count, description="Nodes", layout=layout
+            )
+            scale = Button(description="Scale", layout=layout)
+
+            minimum = IntText(0, description="Minimum", layout=layout)
+            maximum = IntText(0, description="Maximum", layout=layout)
+            adapt = Button(description="Adapt", layout=layout)
+
+            accordion = Accordion(
+                [HBox([request, scale]), HBox([minimum, maximum, adapt])],
+                layout=Layout(min_width="500px"),
+            )
+            accordion.selected_index = None
+            accordion.set_title(0, "Manual Scaling")
+            accordion.set_title(1, "Adaptive Scaling")
+
+            def adapt_cb(b):
+                self.adapt(minimum=minimum.value, maximum=maximum.value)
+                update()
+
+            adapt.on_click(adapt_cb)
+
+            def scale_cb(b):
+                with log_errors():
+                    n = request.value
+                    with ignoring(AttributeError):
+                        self._adaptive.stop()
+                    self.scale(n)
+                    update()
+
+            scale.on_click(scale_cb)
+        else:
+            accordion = HTML("")
+
+        box = VBox([title, HBox([status, accordion]), jupyter, dashboard])
+
+        self._cached_widget = box
+
+        def update():
+            status.value = self._widget_status()
+
+        pc = PeriodicCallback(update, 500, io_loop=self.loop)
+        self.periodic_callbacks["cluster-repr"] = pc
+        pc.start()
+
+        return box
```

Comment:
This is another thing that would be good to fix in the long term but happy to merge as is.

Reply:
You're right -- I literally moved that logic here to extend it: we disambiguate how we print out workers info: `Workers (vCPU)` for CPU clusters, and `Workers (GPUs)` for GPU clusters. This was done (and we def should push it to the `Cluster`) since for GPU clusters both, the memory was showing 0B, and number of Cores was equal to the number of nodes, which is not true for multi-GPU machines/VMs.